### DISCUSSION TITLE
GetThumbnail Performance

### DISCUSSION THREAD

**Discussion by HEIC-to-JPEG-Dev** [dlemstra/Magick.NET#1410] [Help] [answered]
To get the thumbnail of an image I use:

var thumbnail = myMagickImage.Thumbnail(width, height);

Which works fine, and it does some nice things like removing the metadata; however, it is slow and requires the whole image to be read and modified in memory - is there a faster way ? For example, some formats store an embedded thumbnail.

I ask as some of the images I have are huge (5000 x 4000 +)

thanks

**Comment by dlemstra** [comment]
For some formats your might be able to get the information. The `IExifProfile` has thumbnail information. And for raw images (e.g. `dng`, `cr3`) you can use the `ReadThumbnail` property of the `DngReadDefines` to read thumbnail information. But while writing this response I realized that you are missing some information to read the thumbnail so I will need to add something to ImageMagick to make sure you get that information. I will be adding more information in the comments of the `ReadThumbnail` property and on this page: https://imagemagick.org/script/defines.php once that has been added.

**Comment by HEIC-to-JPEG-Dev** [comment, marked as answer]
I've worked through this and I use the EXIF profile to get a fast thumbnail and fall back to resize when not available. This works perfectly, thanks
